src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -311,24 +276,25 @@ def upgrade_app(
             # 3. If existing application is found, perform a few validations and upgrade the application object.
             if show_app_row:
 
-                # We skip comment check here, because prod/pre-existing application objects may not be created by the Snowflake CLI.
-                # Check for the right owner
-                ensure_correct_owner(
-                    row=show_app_row, role=self.app_role, obj_name=self.app_name
-                )
+                install_method.ensure_app_usable(self._na_project, show_app_row)
```

Comment:
at first glance, I'm not a huge fan of this API; it seems like `SameAccountInstallMethod` is kind of just a grab bag of tools that needs to be driven externally in just the right way rather than a class that encapsulates some behaviour of its own. However, I don't want to spend time bikeshedding what's still a net positive for dev velocity, so we should just feel free to revisit this design if it proves to be inflexible in the future

Reply:
You're right; I think we can push the CRUD logic up into this class and potentially merge it with `NativeAppProjectModel` to become the `Application` entity in PDFv2 CC @sfc-gh-gbloom